lib/trino-array/src/main/java/io/trino/array/BlockBigArray.java

```diff
@@ -33,6 +33,11 @@ public BlockBigArray(Block block)
         array = new ObjectBigArray<>(block);
     }
 
+    public long getCapacity()
```

Comment:
Why add this?

Reply:
I've found it useful in other changes to be able to lazily initialize one `XxxBigArray` using the current capacity of another, but those changes are not part of this PR, making this method currently unused (but obvious in it's potential use).

Reply:
Let's add this method once there is usage. Please remove it from this PR

Reply:
Removed

Reply:
I actually think the getCapacity might be quite useful. In quite a few use cases, callers end up tracking this value externally, which is why you don't see a burning need. But if the data structure can report this itself, that would be even better imo.

Reply:
I’m ok waiting for those usages to materialize and adding the getter then. I had a prototype going that was using the method and was trying to get a two-for-one PR merged but that iteration turned out to not improve things as hoped. Should be easy enough for the next person to add them back when they need them.